### ISSUE TITLE
Address and Phone Number Form Fields Missing on Frontend

### ISSUE THREAD

**Issue by qasumitbagthariya** [mailchimp/wordpress#34]
### Describe the bug

The address and phone number form fields are configured and available in the backend settings, but they do not appear on the frontend.

**Expected Behavior:**
Upon enabling the address and phone number fields in the Mailchimp setup under Merge Fields Included, these fields should be visible and accessible on the frontend.

### Steps to Reproduce

1. Navigate to the WordPress Admin Dashboard.
2. Go to "Settings" and select "Mailchimp Setup" from the dropdown menu.
3. Inside the Mailchimp setup, locate and click on "Merge Fields Included".
4. Enable the fields for "Address" and "Phone Number".
5. Save the settings 
6. Visit the frontend of the website.
7. Observe that the address and phone number fields do not appear on the subscription form.

### Screenshots, screen recording, code snippet

https://github.com/mailchimp/wordpress/assets/67687255/9d1bb25a-b3b5-49a5-bbc2-60d2ac1f2e2c

### Environment information

**Testing Environment** 

WordPress: 6.5.4
Theme: Twenty Twenty-Four 1.1
PHP: 8.1.23
Web Server: Nginx 1.20.2
Browser: Chrome 
OS: macOS Ventura 13.3 

### Code of Conduct

- [ ] I agree to follow this project's Code of Conduct

**Comment by dkotter** [comment]
I honestly don't recall specifics at this point but I think it was just around updating the UI of that settings page to match better with WordPress (still looks/feels outdated) as well as looking to move most of the display settings into the block and shortcode, rather than controlling those globally (though may be nice to do both, set some global defaults and then have full control at the block level)

**Comment by MaxwellGarceau** [comment]
@dkotter @jeffpaul I'm about 60% through the rest of the bugs. Want me to tackle the "plugin settings don't reflect Mailchimp account settings" part of this issue? I think I know where in the codebase those changes would be and could get a PR up quickly.

**Comment by dkotter** [comment]
> @dkotter @jeffpaul I'm about 60% through the rest of the bugs. Want me to tackle the "plugin settings don't reflect Mailchimp account settings" part of this issue? I think I know where in the codebase those changes would be and could get a PR up quickly.

That sounds good to me

**Comment by MaxwellGarceau** [comment]
Actually, I believe the "plugin settings don't reflect Mailchimp account settings" is fixed in https://github.com/mailchimp/wordpress/issues/105 🎉. @dkotter can you confirm that the bugs are the same?

I've summarized the other task items here if we wanted to break any of them out of this ticket and work on them in a separate issue.

I can update the documentation, but the other two task items would need a second opinion because they're partially product design questions.

## Task items
- ~~Sync Mailchimp WP settings with Mailchimp account merge field settings~~ - likely handled in https://github.com/mailchimp/wordpress/issues/105
- Add more documentation in readme regarding the connection between Mailchimp WP and Mailchimp account merge field syncronization
- Update the UI of that settings page to match better with WordPress (**Blocker:** need specific updates to make, may already be partially handled with https://github.com/mailchimp/wordpress/issues/62)
- Move most of the display settings into the block and shortcode, rather than controlling those globally
  - Designate the Mailchimp settings page as "Global Settings"

**Comment by iamdharmesh** [comment]
Based on the discussion, this issue seems similar to https://github.com/mailchimp/wordpress/issues/105 and has already been fixed via https://github.com/mailchimp/wordpress/pull/110, where everything works as expected.

However, I'd like to revisit how this currently works and propose an update to simplify it for users. Currently, form fields in the WordPress settings include an "Include?" option. If a field’s "Visibility" is hidden in the Mailchimp account, the option to include that field in the form isn’t shown, meaning a field can only be added to the form after it is made visible from the Mailchimp side.

I recently noticed that the Merge Fields settings page has been updated. It no longer displays the "Visible" option in the edit field, although you can still change field visibility from the form builder.

|![Image](https://github.com/user-attachments/assets/89f0bb2c-c59f-4397-af84-6982ec8054ce)|![Image](https://github.com/user-attachments/assets/65b71bd6-fe18-40b4-b887-f4e19d80f25e)|![Image](https://github.com/user-attachments/assets/ec1b97ef-3224-4f3c-b6c6-67ec153b8d6a)|![Image](https://github.com/user-attachments/assets/6af12523-01ee-4451-a8ca-e60eb0c8e57e)|
|--|--|--|--|
|Fields settings page earlier| Fields settings page now|Edit field popup now (no visibility option)| Field visibility option in form builder|

The visibility setting determines whether a field appears on the form. Since we already have an "Include?" option in WordPress, the field should display based on that value, regardless of the Mailchimp visibility setting (which is more related to the hosted form). Additionally, the "Visible" option was previously part of the fields list table, making it easy to toggle visibility. Now that it’s in the form builder, it might confuse users on how to enable visibility of the field. New fields are added with visibility set to hidden by default, requiring users to enable visibility manually each time.

@jeffpaul @dkotter I suggest updating the current implementation so that the form field's visibility is controlled solely by the "Include?" setting and I’d like to hear your thoughts on this.

cc: @MaxwellGarceau (Tagging you here since you worked on the fix and have context on this. Any input would be appreciated.)

Thanks!

**Comment by dkotter** [comment]
> I suggest updating the current implementation so that the form field's visibility is controlled solely by the "Include?" setting and I’d like to hear your thoughts on this.

So this is a change from how this has always worked so just want to ensure we're thinking about any backwards compatibility issues here (if any). For instance, if we make this change, we need to ensure that we don't suddenly start showing fields within the form that previously weren't showing due to being marked as non-visible on the Mailchimp side.

Assuming we can properly handle that, I like the idea of having more control over things in WordPress, rather than having to potentially change things both on the Mailchimp side and the WordPress side. I'm sure this has caused some confusion for users.

The only other thing to test is if a field is marked as non-visible on the Mailchimp side and we allow users to show this field on the WordPress side, will Mailchimp reject the form entry? Meaning does Mailchimp have any built-in validation to prevent non-visible fields from submitting?

**Comment by iamdharmesh** [comment]
> just want to ensure we're thinking about any backwards compatibility issues here (if any). For instance, if we make this change, we need to ensure that we don't suddenly start showing fields within the form that previously weren't showing due to being marked as non-visible on the Mailchimp side.

Yes, we need to run an upgrade script to mark "include?" false to stop showing it on FE. As currently by default fields are marked as included during initial list setup.

> The only other thing to test is if a field is marked as non-visible on the Mailchimp side and we allow users to show this field on the WordPress side, will Mailchimp reject the form entry? Meaning does Mailchimp have any built-in validation to prevent non-visible fields from submitting?

Field data will get submitted properly without any issue and it reflect correctly on the contact details in Mailchimp side. So, we are good here.
